File: client/types/category.ts
import type { ResponseData } from '~/types/base';

export type CategoryData = {
    level: number;
    id: number;
    created_at: Date;
    slug: string;
    name: string;
    relative_path: string | null;
    url_thumbnail: string | null;
    parent_id: number | null;
    updated_at: Date | null;
    childs?: CategoryData[];
};

export type CategoriesRsp = ResponseData<CategoryData[]>;

File: server/app/schema/category.py
from typing import List, Optional
from pydantic import BaseModel

class CategoryData(BaseModel):
    id: int
    slug: str
    name: str
    url_thumbnail: str
    level: int
    parent_id: int
    relative_path: int
    childs: Optional[List['CategoryData']]


class Category(BaseModel):
    categories: List[CategoryData]


class CategoryResponse(BaseModel):
    status: str
    data: Category
